eureka-client/src/main/java/com/netflix/discovery/shared/EurekaHttpClient.java

```diff
@@ -0,0 +1,57 @@
+package com.netflix.discovery.shared;
+
+import com.netflix.appinfo.InstanceInfo;
+import com.netflix.appinfo.InstanceInfo.InstanceStatus;
+
+/**
+ * Low level Eureka HTTP client API.
+ *
+ * @author Tomasz Bak
+ */
+public interface EurekaHttpClient {
+
+    HttpResponse<Void> register(InstanceInfo info);
+
+    HttpResponse<Void> cancel(String appName, String id);
+
+    HttpResponse<InstanceInfo> sendHeartBeat(String appName, String id, InstanceInfo info, InstanceStatus overriddenStatus);
```

Comment:
This API maps one to one to existing client API, but you are right, we can
do simplification here.

On Tue, Jun 2, 2015 at 5:24 PM, qiangdavidliu <notifications@github.com>
wrote:

> In
> eureka-client/src/main/java/com/netflix/discovery/shared/EurekaHttpClient.java
> <https://github.com/Netflix/eureka/pull/529#discussion_r31584759>:
>
> > +
> > +import com.netflix.appinfo.InstanceInfo;
> > +import com.netflix.appinfo.InstanceInfo.InstanceStatus;
> > +
> > +/**
> > + * Low level Eureka HTTP client API.
> > + *
> > + * @author Tomasz Bak
> > + */
> > +public interface EurekaHttpClient {
> > +
> > +    HttpResponse<Void> register(InstanceInfo info);
> > +
> > +    HttpResponse<Void> cancel(String appName, String id);
> > +
> > +    HttpResponse<InstanceInfo> sendHeartBeat(String appName, String id, InstanceInfo info, InstanceStatus overriddenStatus);
>
> the InstanceInfo should contain the appName and id, so no need to pass
> these as params.
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/Netflix/eureka/pull/529/files#r31584759>.
>